app/models/katello/content_view_environment_content_facet.rb

```diff
@@ -5,5 +5,17 @@ class ContentViewEnvironmentContentFacet < Katello::Model
 
     validates :content_view_environment_id, presence: true
     validates :content_facet_id, presence: true, unless: :new_record?
+    validate :ensure_valid_content_source
+
+    def ensure_valid_content_source
```

Comment:
Relating to my other comments, I think this validation is somehow running too early. Perhaps we're not updating the content source, CV, and LCE at the same time? If we update only the content source but not the LCE, I could see this error happening.

Reply:
That makes sense..I don't remember hitting this issue..Let me spin up some boxes.. :)

Reply:
Yeah, I don't think I took into account if you're _currently in the process of changing_ the content source.

thanks for picking this up while I was gone :)

Reply:
Pushed a change for this. Basically following the order in the UI. Update content_source first and then update CVEs with the validations.